sources/unstructured_data/helpers.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from typing import Any, Dict
+
+from langchain.document_loaders import UnstructuredFileLoader, UnstructuredPDFLoader
+from langchain.indexes import VectorstoreIndexCreator
+
+logging.basicConfig(format="%(asctime)s WARNING: %(message)s", level=logging.WARNING)
+
+filetype_mapper = {
+    ".txt": UnstructuredFileLoader,
+    ".pdf": UnstructuredPDFLoader,
+}
+
+
+def safely_query_index(index: Any, query: str) -> Any:
+    try:
+        answer = index.query(query).strip()
+        return answer
+    except Exception as e:
+        logging.warning(e)
```

Comment:
ohhh we cannot just ignore exceptions. we need to see when they happen and how to handle them. ie. if openAPI call fails we can use tenacity to retry

Reply:
honestly I didn't have any exception, so I don't know how to catch and handle e.g. call fails, I deleted this try-except for now